server/routes/stripe.coffee

```diff
@@ -86,49 +87,57 @@ module.exports.setup = (app) ->
                 subscriptionID: subscriptionID
               }
             })
-            # TODO: load gems from correct Product
-            productGems = 3500
-            if recipient.get('country') is 'brazil'
-              productGems = 1500
-            payment.set 'gems', productGems if invoice.lines.data[0].plan?.id is 'basic'
-
-            payment.save (err) =>
+            Promise.resolve().then ->
+              return null unless invoice.lines.data[0].plan?.id is 'basic'
+              return Product.findBasicSubscriptionForUser(recipient).catch((err) -> err).then (product) ->
+                return {res: 'No subscription product found', code: 500} unless product
+                payment.set 'gems', product.get('gems')
+                null
+            .catch (err) ->
               if err
-                logStripeWebhookError("Save payment error: #{JSON.stringify(err)}")
+                logStripeWebhookError("Fetch product error: #{JSON.stringify(err)}")
                 return res.send(500, '')
-              return res.send(201, '') if invoice.lines.data[0].plan?.id isnt 'basic'
-
-              # Update purchased gems
-              # TODO: is this correct for a resub?
-              Payment.find({recipient: recipient._id, gems: {$exists: true}}).select('gems').exec (err, payments) ->
-                gems = _.reduce payments, ((sum, p) -> sum + (p.get('gems') or 0)), 0
-                purchased = _.clone(recipient.get('purchased'))
-                purchased ?= {}
-                purchased.gems = gems
-                recipient.set('purchased', purchased)
-                recipient.save (err) ->
-                  if err
-                    logStripeWebhookError("Save recipient user error: #{JSON.stringify(err)}")
-                    return res.send(500, '')
-                  return res.send(201, '')
+            .then (err) ->
+              if err
+                logStripeWebhookError("Fetch product error: #{JSON.stringify(err)}")
+                return res.send(500, '')
```

Comment:
I don't think `err` is passed in, nor does it need to be handled here in `then`. If an error is being given as a valid value (by returning a `new Error()`?) then it shouldn't; it should throw it so the `catch`... catches it.